### ISSUE TITLE
Binder asserting during visitModuleName

### ISSUE THREAD

**Issue by rchiodo** [microsoft/pyright#4778]
Pylance telemetry is showing 4 or 5 hundred users on insiders hitting this error:

Error: Debug Failure. False expression.
    at O.visitModuleName (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:96342)
    at O.visit (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:323677)
    at O.visitNode (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:328523)
    at O.walk (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:328402)
    at O.walk (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:74868)
    at <REDACTED: user-file-path> <REDACTED: user-file-path>:1:328487
    at Array.forEach (<anonymous>)
    at O.walkMultiple (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:328466)
    at O.walk (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:328432)
    at O.walk (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:74868)
    at <REDACTED: user-file-path> <REDACTED: user-file-path>:1:328487
    at Array.forEach (<anonymous>)
    at O.walkMultiple (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:328466)
    at O.walk (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:328432)
    at O.walk (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:74868)
    at <REDACTED: user-file-path> <REDACTED: user-file-path>:1:328487
    at Array.forEach (<anonymous>)
    at O.walkMultiple (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:328466)
    at O.walk (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:328432)
    at O.walk (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:74868)
    at O._walkStatementsAndReportUnreachable (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:111605)
    at O.check (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:74464)
    at <REDACTED: user-file-path> <REDACTED: user-file-path>:1:456128
    at i.timeOperation (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:1058118)
    at <REDACTED: user-file-path> <REDACTED: user-file-path>:1:456043
    at t.LogTracker.log (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:1024681)
    at t.SourceFile.check (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:455944)
    at <REDACTED: user-file-path> <REDACTED: user-file-path>:1:358867
    at t.LogTracker.log (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:1024681)
    at j._checkTypes (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:358163)
    at <REDACTED: user-file-path> <REDACTED: user-file-path>:1:358708
    at t.LogTracker.log (<REDACTED: user-file-path> <REDACTED: user-file-path>:1:1024681)

This is NOT happening on the the previous insiders version, so should be something submitted recently.

**Comment by rchiodo** [comment]
Do you know when the bug was introduced? We didn't actually release a stable release today because of the perf problems we've been having. So we may just have to hot fix the prerelease branch.

**Comment by erictraut** [comment]
It was introduced as part of [this change](https://github.com/microsoft/pyright/commit/6910128279b12d193ca908413c172c52b66137d0) on March 5, so it was in last week's pylance prerelease build.

**Comment by heejaechang** [comment]
I found another crash. when user has unfinished import such as

```python
import
```

binder and checker will throw an exception

```
[Error - 3:25:29 AM] An internal error occurred while binding file "Untitled-1": Error: Debug Failure. False expression.
    at Binder.visitModuleName (c:\dd\pyright\packages\vscode-pyright\dist\server.js:33430:28)
    at Binder.visit (c:\dd\pyright\packages\vscode-pyright\dist\server.js:51570:29)
    at Binder.visitNode (c:\dd\pyright\packages\vscode-pyright\dist\server.js:51911:21)
    at Binder.walk (c:\dd\pyright\packages\vscode-pyright\dist\server.js:51895:37)
    at c:\dd\pyright\packages\vscode-pyright\dist\server.js:51903:22
    at Array.forEach (<anonymous>)
    at Binder.walkMultiple (c:\dd\pyright\packages\vscode-pyright\dist\server.js:51901:15)
    at Binder.walk (c:\dd\pyright\packages\vscode-pyright\dist\server.js:51897:18)
    at c:\dd\pyright\packages\vscode-pyright\dist\server.js:51903:22
    ... removed ...
[Error - 3:25:29 AM] An internal error occurred while type checking file "Untitled-1": Error: Debug Failure. False expression.
    at Checker.visitModuleName (c:\dd\pyright\packages\vscode-pyright\dist\server.js:37610:28)
    at Checker.visit (c:\dd\pyright\packages\vscode-pyright\dist\server.js:51570:29)
    at Checker.visitNode (c:\dd\pyright\packages\vscode-pyright\dist\server.js:51911:21)
    at Checker.walk (c:\dd\pyright\packages\vscode-pyright\dist\server.js:51895:37)
    at Checker.walk (c:\dd\pyright\packages\vscode-pyright\dist\server.js:36790:19)
    at c:\dd\pyright\packages\vscode-pyright\dist\server.js:51903:22
    ... removed ...
```

for both of them, it is due to this assert
```ts
const importResult = AnalyzerNodeInfo.getImportInfo(node);
assert(importResult !== undefined);
``` 

ex) https://github.com/microsoft/pyright/blame/main/packages/pyright-internal/src/analyzer/binder.ts#L358 and checker has similar one.

according to the git blame, the assert was there for long time, so it seems something has changed that caused empty module name to no longer has associated `ImportInfo`

not sure whether that's expected or something got deleted by an accident.

**Comment by heejaechang** [comment]
tagging @erictraut

**Comment by erictraut** [comment]
This is included in pyright 1.1.300, which I just published. It will also be included in a future release of pylance.
